src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -52,6 +54,17 @@ public function __construct(ChoiceListFactoryInterface $choiceListFactory = null
                 new DefaultChoiceListFactory()
             )
         );
+
+        // BC
+        if ($this->choiceListFactory instanceof CachingFactoryDecorator) {
+            return;
+        }
+
+        $ref = new \ReflectionMethod($this->choiceListFactory, 'createListFromChoices');
+
+        if (3 !== $ref->getNumberOfParameters() || 'filter' !== $ref->getParameters()[2]->getName()) {
```

Comment:
This check should be improved, see how we do in similar situations in the codebase: we should use `<` instead of `!==` and we shouldn't look for the name of the argument.